src/wallet/bitcoin.ts

```diff
@@ -15,152 +17,163 @@ export interface BitcoinWallet {
+    httpUrl,
+    username,
+    password,
+    network,
+    hdKey,
+    rescan = false
+  }: BitcoindWalletArgs): Promise<BitcoindWallet> {
+    const auth = { username, password };
+
+    const key = !!hdKey
+      ? bip32.fromBase58(hdKey)
+      : bip32.fromSeed(
+          crypto.randomBytes(32),
+          networks[toBitcoinjsLibNetwork(network)]
+        );
+
+    // use HD key fingerprint as wallet name
+    const walletName = key.fingerprint.toString("hex");
+    const walletExists = await axios
+      .post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "listwallets"
+        },
+        { auth }
+      )
+      .then(res => res.data.result.includes(walletName));
+
+    if (!walletExists) {
+      await axios.post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "createwallet",
+          params: [walletName]
+        },
+        { auth }
+      );
     }
 
-    pool.startSync();
-
-    pool.on("tx", (tx: any) => {
-      walletdb.addTX(tx);
-    });
-
-    pool.on("block", (block: any) => {
-      walletdb.addBlock(block);
-      if (block.txs.length > 0) {
-        block.txs.forEach((tx: any) => {
-          walletdb.addTX(tx);
-        });
+    let descriptor = `wpkh(${key.toBase58()}/0h/0h/*h)`;
+    const descriptorInfo = await axios
+      .post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "getdescriptorinfo",
+          params: [descriptor]
+        },
+        { auth }
+      )
+      .then(res => res.data.result);
+    descriptor = `${descriptor}#${descriptorInfo.checksum}`;
+
+    const rpcRequestConfig = {
+      url: `${httpUrl}/wallet/${walletName}`,
+      method: "post" as Method,
+      auth
+    };
+
+    await axios.request({
+      ...rpcRequestConfig,
+      data: {
+        jsonrpc: "1.0",
+        method: "importmulti",
+        params: [[{ desc: descriptor, timestamp: 0, range: 0 }], { rescan }]
       }
     });
 
-    const netAddr = await pool.hosts.addNode(peerUri);
-    const peer = pool.createOutbound(netAddr);
-    pool.peers.add(peer);
-
-    return new InMemoryBitcoinWallet(
-      parsedNetwork,
-      walletdb,
-      pool,
-      chain,
-      wallet
-    );
+    return new BitcoindWallet(rpcRequestConfig);
   }
 
-  private constructor(
-    public readonly network: any,
-
-    private readonly walletdb: any,
-    private readonly pool: any,
-
-    private readonly chain: any,
-    private readonly wallet: any
-  ) {}
+  private constructor(private rpcRequestConfig: AxiosRequestConfig) {}
 
   public async getBalance(): Promise<number> {
-    const balance = await this.wallet.getBalance();
-    // TODO: Balances stay unconfirmed, try to use bcoin.SPVNode (and set node.http to undefined) see if it catches the confirmations
-    const amount = new Amount(balance.toJSON().unconfirmed, "sat");
-    return amount.toBTC(true);
+    const res = await axios.request({
+      ...this.rpcRequestConfig,
```

Comment:
You can just make an axios instance instead of doing this :)

Check what we do in the `Cnd` class.